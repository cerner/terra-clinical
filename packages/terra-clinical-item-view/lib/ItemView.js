'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _terraClinicalItemDisplay = require('terra-clinical-item-display');

var _terraClinicalItemDisplay2 = _interopRequireDefault(_terraClinicalItemDisplay);

require('terra-base/lib/baseStyles');

require('./ItemView.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  /**
   * The column layout in which to present the displays.
   */
  layout: _propTypes2.default.oneOf(['oneColumn', 'twoColumns']),
  /**
   * The text color emphasis when using two columns.
   */
  textEmphasis: _propTypes2.default.oneOf(['default', 'start']),
  /**
   * Whether or not all text on the view should be truncated.
   */
  isTruncated: _propTypes2.default.bool,
  /**
   * The vertical alignment of the start and end accesories.
   */
  accessoryAlignment: _propTypes2.default.oneOf(['alignTop', 'alignCenter']),
  /**
   * The react element to be placed in the start aligned accessory position.
   */
  startAccessory: _propTypes2.default.element,
  /**
   * The react element to be placed in the end aligned accessory position.
   */
  endAccessory: _propTypes2.default.element,
  /**
   * An array of react display elements to be presented.
   */
  displays: _propTypes2.default.arrayOf(_propTypes2.default.element),
  /**
   * The react element for the comment element.
   */
  comment: _propTypes2.default.element
};

var defaultProps = {
  layout: 'oneColumn',
  textEmphasis: 'default',
  isTruncated: false,
  accessoryAlignment: 'alignCenter',
  startAccessory: undefined,
  endAccessory: undefined,
  displays: [],
  comment: undefined
};

var ItemView = function (_React$Component) {
  _inherits(ItemView, _React$Component);

  function ItemView() {
    _classCallCheck(this, ItemView);

    return _possibleConstructorReturn(this, (ItemView.__proto__ || Object.getPrototypeOf(ItemView)).apply(this, arguments));
  }

  _createClass(ItemView, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          layout = _props.layout,
          textEmphasis = _props.textEmphasis,
          isTruncated = _props.isTruncated,
          accessoryAlignment = _props.accessoryAlignment,
          startAccessory = _props.startAccessory,
          endAccessory = _props.endAccessory,
          displays = _props.displays,
          comment = _props.comment,
          customProps = _objectWithoutProperties(_props, ['layout', 'textEmphasis', 'isTruncated', 'accessoryAlignment', 'startAccessory', 'endAccessory', 'displays', 'comment']);

      var viewClassNames = (0, _classnames2.default)(['terraClinical-ItemView', { 'terraClinical-ItemView--isTruncated': isTruncated }, _defineProperty({}, 'terraClinical-ItemView--' + layout, layout), customProps.className]);

      return _react2.default.createElement(
        'div',
        _extends({}, customProps, { className: viewClassNames }),
        ItemView.renderAccessory(startAccessory, accessoryAlignment),
        _react2.default.createElement(
          'div',
          { className: 'terraClinical-ItemView-body' },
          ItemView.renderRows(displays, layout, textEmphasis),
          comment
        ),
        ItemView.renderAccessory(endAccessory, accessoryAlignment)
      );
    }
  }], [{
    key: 'renderAccessory',
    value: function renderAccessory(accessory, accessoryAlignment) {
      var accessoryClassNames = (0, _classnames2.default)(['terraClinical-ItemView-accessory', _defineProperty({}, 'terraClinical-ItemView-accessory--' + accessoryAlignment, accessoryAlignment)]);

      var accessorySection = void 0;
      if (accessory) {
        accessorySection = _react2.default.createElement(
          'div',
          { className: accessoryClassNames },
          accessory
        );
      }
      return accessorySection;
    }
  }, {
    key: 'renderRows',
    value: function renderRows(displays, layout, emphasis) {
      if (displays === null || displays === undefined || !displays.length) {
        return undefined;
      }

      var displayGroups = [];
      var displaysSlice = displays.slice(0, 8);
      var spliceValue = layout === 'twoColumns' ? 2 : 1;

      while (displaysSlice.length) {
        displayGroups.push(displaysSlice.splice(0, spliceValue));
      }

      return _react2.default.createElement(
        'div',
        { className: 'terraClinical-ItemView-rowContainer' },
        displayGroups.map(function (group, index) {
          var row = ItemView.renderRow(group, index, displayGroups.length, emphasis);
          return row;
        })
      );
    }
  }, {
    key: 'renderRow',
    value: function renderRow(row, rowIndex, rowCount, emphasis) {
      var rowKey = rowIndex;
      return _react2.default.createElement(
        'div',
        { className: 'terraClinical-ItemView-row', key: rowKey },
        row.map(function (display, contentIndex) {
          var contentKey = contentIndex;
          var contentClasses = ItemView.classesForContent(rowIndex, rowCount, contentIndex, emphasis);
          return _react2.default.createElement(
            'div',
            { className: contentClasses, key: contentKey },
            display
          );
        })
      );
    }
  }, {
    key: 'classesForContent',
    value: function classesForContent(rowIndex, rowCount, contentIndex, emphasis) {
      var classes = void 0;
      if (emphasis === 'start') {
        classes = ItemView.startEmphasisContentClassesFromIndexes(rowIndex, rowCount, contentIndex);
      } else {
        classes = ItemView.defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount);
      }
      return ['terraClinical-ItemView-content'].concat(classes).join(' ');
    }
  }, {
    key: 'defaultEmphasisContentClassesFromIndexes',
    value: function defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount) {
      var contentSize = 'terraClinical-ItemView-content--primarySize';
      var contentColor = 'terraClinical-ItemView-content--primaryColor';

      if (rowIndex > 0) {
        contentSize = 'terraClinical-ItemView-content--secondarySize';
      }

      if (rowCount === 2 && rowIndex === 1) {
        contentColor = 'terraClinical-ItemView-content--secondaryColor';
      } else if (rowIndex >= 2) {
        contentColor = 'terraClinical-ItemView-content--secondaryColor';
      }

      return [contentSize, contentColor];
    }
  }, {
    key: 'startEmphasisContentClassesFromIndexes',
    value: function startEmphasisContentClassesFromIndexes(rowIndex, rowCount, contentIndex) {
      if (contentIndex === 1) {
        return ['terraClinical-ItemView-content--secondarySize', 'terraClinical-ItemView-content--secondaryColor'];
      }

      return ItemView.defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount);
    }
  }]);

  return ItemView;
}(_react2.default.Component);

ItemView.propTypes = propTypes;
ItemView.defaultProps = defaultProps;
ItemView.Display = _terraClinicalItemDisplay2.default;
ItemView.Comment = _terraClinicalItemDisplay2.default.Comment;

exports.default = ItemView;