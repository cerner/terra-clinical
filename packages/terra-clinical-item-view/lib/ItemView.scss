@import '~terra-mixins';
@import './variables';
@import './mixins';

// Temporary until it is added to application.
.terraClinical-ItemView {
  word-break: break-word;
  word-wrap: break-word;
}

.terraClinical-ItemView {
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  width: 100%;
}

// Accessory

.terraClinical-ItemView-accessory {
  // Accessory is of fixed size and does not shrink or grow.
  flex: 0 0 auto;
  margin: 5px;
  max-height: 60px;
  max-width: 60px;
  min-width: 0;
  overflow: hidden;
}

.terraClinical-ItemView-accessory--alignTop {
  align-self: flex-start;
}

.terraClinical-ItemView-accessory--alignCenter {
  align-self: center;
}

// Content Layout

.terraClinical-ItemView-body {
  flex: 1 1 auto;
  margin: 5px;
  min-width: 0;
  overflow: hidden; // VERY IMPORTANT FOR IE10
}

.terraClinical-ItemView-row {
  display: flex;
  width: 100%;
}

.terraClinical-ItemView-content {
  align-items: center;
  display: flex;
  overflow: hidden; // VERY IMPORTANT FOR IE10
}

.terraClinical-ItemView-text {
  flex: 1 1 auto;
  min-width: 10px; // Need min-width, as IE10 will completely squish display
}

.terraClinical-ItemView-text--inlineIcon {
  @include terra-padding-end(5px);
  flex: 0 0 auto;
  height: 20px;
  max-width: 20px;
  overflow: hidden;
}

.terraClinical-ItemView-text--inlineIcon > * {
  font-size: 1rem;
}

// Truncation

.terraClinical-ItemView--isTruncated .terraClinical-ItemView-text,
.terraClinical-ItemView-text--isTruncated {
  @include terra-clinical-text-truncate;
}

// Layouts

.terraClinical-ItemView--oneColumn {
  .terraClinical-ItemView-rowContainer {
    .terraClinical-ItemView-row {
      flex-flow: column nowrap;
      justify-content: flex-start;
    }

    .terraClinical-ItemView-content {
      width: 100%;
    }
  }
}

.terraClinical-ItemView--twoColumns {
  .terraClinical-ItemView-content:nth-child(odd) {
    flex: 1 1 auto;
    text-align: left;
  }

  .terraClinical-ItemView-content:nth-child(even) {
    @include terra-padding-start(5px);

    flex: 1 0 auto;
    justify-content: flex-end;

    // Set a max-width and disable flexbox-squishing on the right-most displays. This will ensure that they aren't
    // prematurely squished and maximize the available space for all labels.
    max-width: 60%;

    .terraClinical-ItemView-text {
      flex: 0 1 auto;
      min-width: 10px; // Need min-width, as IE10 will completely squish display
    }
  }
}

// We need to style content that appears out of the rowContainer (i.e. the comment).
.terraClinical-ItemView-content {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    @include terra-clinical-text-body;
    color: $terra-clinical-item-view-content-secondary-color;
  }
}

//Emphasis Content Styles
.terraClinical-ItemView-content--primaryColor {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    color: $terra-clinical-item-view-content-primary-color;
  }
}

.terraClinical-ItemView-content--primarySize {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    @include terra-clinical-tiny-heading;
  }
}

.terraClinical-ItemView-content--secondaryColor {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    color: $terra-clinical-item-view-content-secondary-color;
  }
}

.terraClinical-ItemView-content--secondarySize {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    @include terra-clinical-text-body;
  }
}

// Text Styles

.terraClinical-ItemView-text--secondary {
  color: $terra-clinical-item-view-text-secondary-color !important;
}

.terraClinical-ItemView-text--attention {
  color: $terra-clinical-item-view-text-attention-color !important;
  font-weight: $terra-clinical-item-view-text-attention-weight !important;
}

.terraClinical-ItemView-text--strikeThrough {
  text-decoration: line-through !important;
}
