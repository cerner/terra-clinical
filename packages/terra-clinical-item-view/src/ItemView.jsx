import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ItemDisplay from 'terra-clinical-item-display';
import 'terra-base/lib/baseStyles';
import styles from './ItemView.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The column layout in which to present the displays.
   */
  layout: PropTypes.oneOf(['oneColumn', 'twoColumns']),
  /**
   * The text color emphasis when using two columns.
   */
  textEmphasis: PropTypes.oneOf(['default', 'start']),
  /**
   * Whether or not all text on the view should be truncated.
   */
  isTruncated: PropTypes.bool,
  /**
   * The vertical alignment of the start and end accesories.
   */
  accessoryAlignment: PropTypes.oneOf(['alignTop', 'alignCenter']),
  /**
   * The react element to be placed in the start aligned accessory position.
   */
  startAccessory: PropTypes.element,
  /**
   * The react element to be placed in the end aligned accessory position.
   */
  endAccessory: PropTypes.element,
  /**
   * An array of react display elements to be presented.
   */
  displays: PropTypes.arrayOf(PropTypes.element),
  /**
   * The react element for the comment element.
   */
  comment: PropTypes.element,
};

const defaultProps = {
  layout: 'oneColumn',
  textEmphasis: 'default',
  isTruncated: false,
  accessoryAlignment: 'alignCenter',
  startAccessory: undefined,
  endAccessory: undefined,
  displays: [],
  comment: undefined,
};

class ItemView extends React.Component {

  static renderAccessory(accessory, accessoryAlignment) {
    let accessorySection;
    if (accessory) {
      const accessoryClassNames = cx(
        'accessory',
        { 'accessory-align-center': accessoryAlignment === 'alignCenter' },
        { 'accessory-align-top': accessoryAlignment === 'alignTop' },
      );

      accessorySection = (
        <div className={accessoryClassNames}>
          {accessory}
        </div>
      );
    }
    return accessorySection;
  }

  static renderRows(displays, layout, emphasis) {
    if (displays === null || displays === undefined || !displays.length) {
      return undefined;
    }

    const displayGroups = [];
    const displaysSlice = displays.slice(0, 8);
    const spliceValue = layout === 'twoColumns' ? 2 : 1;

    while (displaysSlice.length) {
      displayGroups.push(displaysSlice.splice(0, spliceValue));
    }

    return (
      <div className={cx('row-container')}>
        {displayGroups.map((group, index) => {
          const row = ItemView.renderRow(group, index, displayGroups.length, emphasis);
          return row;
        })}
      </div>
    );
  }

  static renderRow(row, rowIndex, rowCount, emphasis) {
    const rowKey = rowIndex;
    return (
      <div className={cx('row')} key={rowKey}>
        {row.map((display, contentIndex) => {
          const contentKey = contentIndex;
          const contentClasses = ItemView.classesForContent(rowIndex, rowCount, contentIndex, emphasis);
          return (
            <div className={cx(contentClasses)} key={contentKey}>
              {display}
            </div>
          );
        })}
      </div>
    );
  }

  static classesForContent(rowIndex, rowCount, contentIndex, emphasis) {
    let classes;
    if (emphasis === 'start') {
      classes = ItemView.startEmphasisContentClassesFromIndexes(rowIndex, rowCount, contentIndex);
    } else {
      classes = ItemView.defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount);
    }
    return ['content'].concat(classes);
  }

  static defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount) {
    let contentSize = 'content-primary-size';
    let contentColor = 'content-primary-color';

    if (rowIndex > 0) {
      contentSize = 'content-secondary-size';
    }

    if (rowCount === 2 && rowIndex === 1) {
      contentColor = 'content-secondary-color';
    } else if (rowIndex >= 2) {
      contentColor = 'content-secondary-color';
    }

    return [contentSize, contentColor];
  }

  static startEmphasisContentClassesFromIndexes(rowIndex, rowCount, contentIndex) {
    if (contentIndex === 1) {
      return ['content-secondary-size', 'content-secondary-color'];
    }

    return ItemView.defaultEmphasisContentClassesFromIndexes(rowIndex, rowCount);
  }

  render() {
    const { layout,
            textEmphasis,
            isTruncated,
            accessoryAlignment,
            startAccessory,
            endAccessory,
            displays,
            comment,
            ...customProps } = this.props;

    const viewClassNames = cx([
      'item-view',
      { 'is-truncated': isTruncated },
      { 'one-column': layout === 'oneColumn' },
      { 'two-columns': layout === 'twoColumns' },
      customProps.className,
    ]);

    return (
      <div {...customProps} className={viewClassNames}>
        {ItemView.renderAccessory(startAccessory, accessoryAlignment)}
        <div className={cx('body')}>
          {ItemView.renderRows(displays, layout, textEmphasis)}
          {comment}
        </div>
        {ItemView.renderAccessory(endAccessory, accessoryAlignment)}
      </div>
    );
  }
}

ItemView.propTypes = propTypes;
ItemView.defaultProps = defaultProps;
ItemView.Display = ItemDisplay;
ItemView.Comment = ItemDisplay.Comment;

export default ItemView;
