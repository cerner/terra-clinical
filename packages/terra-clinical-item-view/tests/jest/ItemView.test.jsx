import React from 'react';
import ItemView from '../../lib/ItemView';

// Snapshot Tests
it('should render a default component', () => {
  const itemView = shallow(<ItemView />);
  expect(itemView).toMatchSnapshot();
});

it('should render with 1 display', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const displays = [display1];
  const itemView = shallow(<ItemView displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render with 2 displays', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const displays = [display1, display2];
  const itemView = shallow(<ItemView displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render with 3 displays', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const display3 = <ItemView.Display text="display 3" />;
  const displays = [display1, display2, display3];
  const itemView = shallow(<ItemView displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render with a display and graphic', () => {
  const testElement = <img alt="Graphic" />;
  const display1 = <ItemView.Display text="display 1" icon={testElement} />;
  const displays = [display1];
  const itemView = shallow(<ItemView displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render truncated display', () => {
  const display1 = <ItemView.Display text="display1display1display1display1display1display1display1display1" isTruncated />;
  const displays = [display1];
  const itemView = shallow(<ItemView displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render a comment', () => {
  const comment = <ItemView.Comment text="comment" textStyle="attention" isTruncated={false} />;
  const params = {
    layout: 'oneColumn',
    textEmphasis: 'default',
    displays: [],
    comment,
  };
  const itemView = shallow(<ItemView {...params} />);
  expect(itemView).toMatchSnapshot();
});

it('should render 1 start theme display', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const displays = [display1];
  const itemView = shallow(<ItemView textEmphasis="start" displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render 2 start theme displays', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const displays = [display1, display2];
  const itemView = shallow(<ItemView textEmphasis="start" displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render 3 start theme displays', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const display3 = <ItemView.Display text="display 3" />;
  const displays = [display1, display2, display3];
  const itemView = shallow(<ItemView textEmphasis="start" displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render a start accessory', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    textEmphasis: 'default',
    displays: [],
    startAccessory: testElement,
  };
  const itemView = shallow(<ItemView {...params} />);
  expect(itemView).toMatchSnapshot();
});

it('should render a end accessory', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    textEmphasis: 'default',
    displays: [],
    endAccessory: testElement,
  };
  const itemView = shallow(<ItemView {...params} />);
  expect(itemView).toMatchSnapshot();
});

it('should render an accessory top aligned', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    textEmphasis: 'default',
    displays: [],
    startAccessory: testElement,
    accessoryAlignment: 'alignTop',
  };
  const itemView = shallow(<ItemView {...params} />);
  expect(itemView).toMatchSnapshot();
});

it('should render an accessory center aligned', () => {
  const testElement = <img alt="Graphic" />;
  const params = {
    layout: 'oneColumn',
    textEmphasis: 'default',
    displays: [],
    startAccessory: testElement,
    accessoryAlignment: 'alignCenter',
  };
  const itemView = shallow(<ItemView {...params} />);
  expect(itemView).toMatchSnapshot();
});

it('should render one column', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const displays = [display1, display2];
  const itemView = shallow(<ItemView layout="oneColumn" displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render two columns with 8 displays', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const display3 = <ItemView.Display text="display 3" />;
  const display4 = <ItemView.Display text="display 4" />;
  const display5 = <ItemView.Display text="display 5" />;
  const display6 = <ItemView.Display text="display 6" />;
  const display7 = <ItemView.Display text="display 7" />;
  const display8 = <ItemView.Display text="display 8" />;
  const display9 = <ItemView.Display text="display 9" />;

  const displays = [display1, display2, display3, display4, display5, display6, display7, display8, display9];
  const itemView = shallow(<ItemView layout="twoColumns" displays={displays} />);
  expect(itemView).toMatchSnapshot();
});

it('should render two columns with odd number of displays', () => {
  const display1 = <ItemView.Display text="display 1" />;
  const display2 = <ItemView.Display text="display 2" />;
  const display3 = <ItemView.Display text="display 3" />;
  const displays = [display1, display2, display3];
  const itemView = shallow(<ItemView layout="twoColumns" displays={displays} />);
  expect(itemView).toMatchSnapshot();
});
