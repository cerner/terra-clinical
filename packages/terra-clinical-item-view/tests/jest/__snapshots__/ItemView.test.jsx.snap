// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 start theme display 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 2 start theme displays 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 3 start theme displays 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render a comment 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <ItemComment
      isTruncated={false}
      text="comment"
      textStyle="attention"
    />
  </div>
</div>
`;

exports[`should render a default component 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  />
</div>
`;

exports[`should render a end accessory 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  />
  <div
    className="accessory accessory-align-center"
  >
    <img
      alt="Graphic"
    />
  </div>
</div>
`;

exports[`should render a start accessory 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="accessory accessory-align-center"
  >
    <img
      alt="Graphic"
    />
  </div>
  <div
    className="body"
  />
</div>
`;

exports[`should render an accessory center aligned 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="accessory accessory-align-center"
  >
    <img
      alt="Graphic"
    />
  </div>
  <div
    className="body"
  />
</div>
`;

exports[`should render an accessory top aligned 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="accessory accessory-align-top"
  >
    <img
      alt="Graphic"
    />
  </div>
  <div
    className="body"
  />
</div>
`;

exports[`should render one column 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render truncated display 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={true}
            text="display1display1display1display1display1display1display1display1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render two columns with 8 displays 1`] = `
<div
  className="item-view two-columns"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 3"
          />
        </div>
        <div
          className="content content-secondary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 4"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 5"
          />
        </div>
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 6"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 7"
          />
        </div>
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 8"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render two columns with odd number of displays 1`] = `
<div
  className="item-view two-columns"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with 1 display 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with 2 displays 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with 3 displays 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-primary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 2"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="content content-secondary-size content-secondary-color"
        >
          <ItemDisplay
            isTruncated={false}
            text="display 3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with a display and graphic 1`] = `
<div
  className="item-view one-column"
>
  <div
    className="body"
  >
    <div
      className="row-container"
    >
      <div
        className="row"
      >
        <div
          className="content content-primary-size content-primary-color"
        >
          <ItemDisplay
            icon={
              <img
                alt="Graphic"
              />
            }
            isTruncated={false}
            text="display 1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
